aiida_quantumespresso/calculations/pp.py

```diff
@@ -1,27 +1,240 @@
+        ('INPUTPP', 'plot_num', None),
+        ('PLOT', 'fileout', _FILEOUT),
+        ('PLOT', 'iflag', None),
+        ('PLOT', 'output_format', None)
     ]
 
+    _retrieve_singlefile_list = []
+
+    _DEFAULT_INPUT_FILE = 'aiida.in'
+    _DEFAULT_OUTPUT_FILE = 'aiida.out'
+
     @classmethod
     def define(cls, spec):
         # yapf: disable
         super(PpCalculation, cls).define(spec)
-        spec.input('parent_folder', valid_type=(RemoteData, FolderData),
+        spec.input('parent_folder', valid_type=(orm.RemoteData, orm.FolderData), required=True,
             help='Output folder of a completed `PwCalculation`')
+        spec.input('plot_number', valid_type=orm.Int, required=True,
+            help='Plot number to compute')
+        spec.input('plot_dimension', valid_type=orm.Int, required=True,
+            help='Plot dimension to compute')
+        spec.input('parameters', valid_type=orm.Dict, required=False,
+            help='Use a node that specifies the input parameters for the namelists')
+        spec.input('settings', valid_type=orm.Dict, required=False,
+            help='Use an additional node for special settings')
+        spec.input('metadata.options.input_filename', valid_type=six.string_types, default=cls._DEFAULT_INPUT_FILE)
+        spec.input('metadata.options.output_filename', valid_type=six.string_types, default=cls._DEFAULT_OUTPUT_FILE)
+        spec.input('metadata.options.parser_name', valid_type=six.string_types, default='quantumespresso.pp')
+        spec.input('metadata.options.withmpi', valid_type=bool, default=True)
+
+        spec.output('output_parameters', valid_type=orm.Dict)
+        spec.output('output_data', valid_type=(orm.XyData, orm.ArrayData))
+
+        spec.exit_code(400, 'ERROR_NO_RETRIEVED_FOLDER',
+            message='The retrieved folder data node could not be accessed.')
+        spec.exit_code(401, 'ERROR_OUTPUT_STDOUT_MISSING',
+            message='The retrieved folder did not contain the required stdout output file.')
+        spec.exit_code(402, 'ERROR_OUTPUT_DATAFILE_MISSING',
+            message='The retrieved folder did not contain the required formatted data output file.')
+        spec.exit_code(403, 'ERROR_PARENT_XML_MISSING',
+            message='The parent folder did not contain the required XML output file.')
+        spec.exit_code(410, 'ERROR_OUTPUT_STDOUT_NOT_READ',
+            message='The stdout output file could not be read.')
+        spec.exit_code(411, 'ERROR_OUTPUT_DATAFILE_NOT_READ',
+            message='The formatted data output file could not be read.')
+        spec.exit_code(420, 'ERROR_OUTPUT_STDOUT_INCOMPLETE',
+            message='The stdout output file was incomplete.')
+        spec.exit_code(430, 'ERROR_UNSUPPORTED_DATAFILE_FORMAT',
+            message='The data file format is not supported by the parser')
+        spec.exit_code(500, 'ERROR_UNEXPECTED_PARSER_EXCEPTION',
+            message='The parser raised an unexpected exception.')
```

Comment:
I homogenized the exit codes across all calculation jobs in PR #479 . Please take a look at, especially `PwCalculation` and match exit code statuses and names.

Reply:
Matched up now.